keel-sql/src/main/resources/db/changelog/20210226-generate-event-columns.yml

```diff
@@ -0,0 +1,74 @@
+databaseChangeLog:
+- changeSet:
+    id: generate-event-timestamp
+    author: fletch
+    changes:
+    - dropIndex:
+        tableName: event
+        indexName: event_uid_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_ref_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_application_timestamp_idx
+    - dropColumn:
+        tableName: event
+        columnName: application
+    - sql:
+        sql: |
+          alter table event
+          add column ref_gen varchar(255) generated always as (json ->> '$.ref')
+          not null
+          after scope;
+    - sql:
+        sql: |
+          alter table event
+          add column application varchar(255) generated always as (json ->> '$.application')
+          not null
+          after ref_gen;
```

Comment:
This column definition isn't changing, I'm just repositioning it in the table

Reply:
My only theoretical concern is how long the event table will stay locked during the migration. Given that it's a generated column being dropped and re-added afterwards, I assume mysql is smart enough that this change happens pretty quickly?

Reply:
I tested it out locally with a dump of prod data and it takes barely any time at all. I was pretty surprised.